bundle/Core/Search/Elasticsearch/Query/Common/CriterionVisitor/Tags.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Netgen\TagsBundle\Core\Search\Elasticsearch\Query\Common\CriterionVisitor;
+
+use Ibexa\Contracts\Core\Persistence\Content\Type\Handler;
+use Ibexa\Contracts\Core\Repository\Values\Content\Query\Criterion;
+use Ibexa\Contracts\Elasticsearch\Query\CriterionVisitor;
+use Ibexa\Core\Search\Common\FieldNameResolver;
+
+use function array_merge;
+
+abstract class Tags implements CriterionVisitor
+{
+    private FieldNameResolver $fieldNameResolver;
+
+    /**
+     * For tag-queries which aren't field-specific.
+     */
+    private Handler $contentTypeHandler;
+
+    /**
+     * Identifier of the field type that criterion can handle.
+     */
+    private string $fieldTypeIdentifier;
+
+    /**
+     * Name of the field type's indexed field that criterion can handle.
+     */
+    private string $fieldName;
+
+    public function __construct(FieldNameResolver $fieldNameResolver, Handler $contentTypeHandler, string $fieldTypeIdentifier, string $fieldName)
```

Comment:
Since we are on PHP 8.1 here, we can use constructor property promotion.

Reply:
fixed in f4b8cce7ea0a8525641b58a5d8cd5841bd2a9ac7